modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/validation/EqualsAndHashCodeModelValidator.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.extension.internal.loader.validation;
+
+import static org.mule.runtime.extension.api.util.ExtensionMetadataTypeUtils.getType;
+import static java.lang.String.format;
+import static java.lang.reflect.Modifier.isAbstract;
+import static org.mule.runtime.module.extension.internal.loader.validation.ModelValidationUtils.isASTMode;
+import static org.mule.runtime.module.extension.internal.loader.validation.ModelValidationUtils.isCompiletime;
+
+import org.mule.metadata.api.model.ObjectType;
+import org.mule.metadata.api.visitor.MetadataTypeVisitor;
+import org.mule.runtime.api.meta.NamedObject;
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this unused import 'org.mule.runtime.api.meta.NamedObject'. ([squid:UselessImportCheck](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AUselessImportCheck))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7618&issues=AWiAl_QyotIXnmsUW-bQ&open=AWiAl_QyotIXnmsUW-bQ)

[ISSUE_KEY]: AWiAl_QyotIXnmsUW-bQ